src/main/java/openfasttrack/core/LinkedSpecificationItem.java

```diff
@@ -157,6 +157,30 @@ public void addCoveredArtifactType(final String artifactType)
     }
 
     /**
+     * Get a list of all artifact types that have unwanted coverage.
+     * 
+     * @return list of over-covered artifact types.
+     */
+    public List<String> getOverCoveredArtifactTypes()
+    {
+        final List<String> overCovered = new ArrayList<>(getCoveredArtifactTypes());
+        overCovered.removeAll(getNeedsArtifactTypes());
+        return overCovered;
+    }
```

Comment:
Here the streams variant would have been harder to read. Due to the low number of items probably also slower. Will be replaced by a bit field in performance enhanced version anyway.